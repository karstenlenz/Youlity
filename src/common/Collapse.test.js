import { render, screen } from '@testing-library/react'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import Collapse from './Collapse'

describe('The Collapse component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Router>
        <Collapse headline="test Headline" headlineOpen="collapse is open">
          Collapse content lorem ipsum
        </Collapse>
      </Router>
    )
    expect(container).toMatchSnapshot()
  })

  it('hides the content at first', () => {
    render(
      <Router>
        <Collapse headline="test Headline" headlineOpen="collapse is open">
          Collapse content
        </Collapse>
      </Router>
    )
    expect(screen.queryByText('Collapse content')).not.toBeVisible()
  })

  it('shows the content after a click', () => {
    render(
      <Router>
        <Collapse headline="test Headline" headlineOpen="collapse is open">
          Collapse content
        </Collapse>
      </Router>
    )

    screen.getByText('test Headline').click()
    expect(screen.getByText('Collapse content')).toBeVisible()
  })
})
