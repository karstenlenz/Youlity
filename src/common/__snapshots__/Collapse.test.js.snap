// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Collapse component renders correctly 1`] = `
<div>
  <section
    class="Collapse__CollapseWrapper-sc-1tgbx39-0 imOYVX"
  >
    <div
      class="Collapse__CollapseHeader-sc-1tgbx39-1 hKtGZf"
    >
      test Headline
      <img
        alt="auf/zuklappen"
        class="Collapse__CollapseIcon-sc-1tgbx39-2 bFSuxH"
        src="/img/expand.svg"
      />
    </div>
    <div
      class="Collapse__CollapseContent-sc-1tgbx39-3 domdvA"
    >
      Collapse content lorem ipsum
    </div>
  </section>
</div>
`;
